src/console_cmds.cpp

```diff
@@ -919,6 +919,12 @@ DEF_CONSOLE_CMD(ConNetworkConnect)
 	IConsolePrintF(CC_DEFAULT, "Connecting to %s...", ip);
 	if (company != nullptr) {
 		join_as = (CompanyID)atoi(company);
+		IConsolePrintF(CC_DEFAULT, "Connecting as company id %s ...", company);
+		/* Check if we have a valid company id! */
+		if(!Company::IsValidID(join_as) && join_as != COMPANY_SPECTATOR){
+			IConsolePrintF(CC_ERROR, "Company does not exist. Company-id must be between 1 and %d.", MAX_COMPANIES);
+			return true;
+		}
 		IConsolePrintF(CC_DEFAULT, "    company-no: %d", join_as);
```

Comment:
I'd remove this in favour of the PrintF you've added (but perhaps your PrintF should use the actual parsed number rather than the string?)

Reply:
I'm sorry, but I'm confused as to what you mean, what should be removed in favor of which PrintF?

Reply:
The PrintF added on line 922

Reply:
So replace 922 printf with the error printf and add the company id to it?

Reply:
What I mean is remove this line entirely, but use the join_as number in your CC_DEFAULT PrintF to ensure the string -> int parse has worked as expected. Neither are error printfs.

Reply:
I'm not sure if this is what you wanted, but i made changes in the newest commit, if you could tell me if that's what you meant.

Reply:
I think what LordAro meant is that line 922 in the new code tells the user that we are going to join as a particular company, but then line 928 (also new code line numbering, but from the old code) effectively gives you the same information. So when you join correctly as company N you get the following two lines:
Connecting as company id N ...
    company-no: N
Both lines give you the same information, the line you added is more user friendly so I suggest removing the second line of output, i.e. remove the line this comment is below.